core/templates/dev/head/pages/exploration_editor/ExplorationSaveAndPublishButtons.js

```diff
@@ -0,0 +1,69 @@
+oppia.controller('ExplorationSaveAndPublishButtons', [
+  '$scope', '$http', '$rootScope', '$window',
+  'changeListService', 'editabilityService',
+  'explorationRightsService', 'explorationWarningsService',
+  'explorationSaveService',
+  function(
+      $scope, $http, $rootScope, $window,
+      changeListService, editabilityService,
+      explorationRightsService, explorationWarningsService,
+      explorationSaveService) {
+    $scope.isSaveInProgress = false;
+
+    $scope.isPrivate = function() {
+      return explorationRightsService.isPrivate();
+    };
+
+    $scope.isExplorationLockedForEditing = function() {
+      return changeListService.isExplorationLockedForEditing();
+    };
+
+    $scope.isEditableOutsideTutorialMode = function() {
+      return editabilityService.isEditableOutsideTutorialMode();
+    };
+
+    $scope.countWarnings = function() {
+      return explorationWarningsService.countWarnings();
+    };
+
+    $scope.discardChanges = function() {
+      explorationSaveService.discardChanges();
+    };
+
+    $scope.getChangeListLength = function() {
+      return changeListService.getChangeList().length;
+    };
+
+    $scope.isExplorationSaveable = function() {
+      return explorationSaveService.isExplorationSaveable();
+    };
+
+    $scope.isPublishModalOpening = function() {
```

Comment:
I think this should be a variable, like saveIsInProgress. See my earlier comment.